parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java

```diff
@@ -670,7 +673,7 @@ private static void copy(SeekableInputStream from, PositionOutputStream to,
   public void end(Map<String, String> extraMetaData) throws IOException {
     state = state.end();
     LOG.debug("{}: end", out.getPos());
-    ParquetMetadata footer = new ParquetMetadata(new FileMetaData(schema, extraMetaData, Version.FULL_VERSION), blocks);
+    this.footer = new ParquetMetadata(new FileMetaData(schema, extraMetaData, Version.FULL_VERSION), blocks);
```

Comment:
(nit, optional) "this." is superfluous, but doesn't hurt either.

Reply:
Using `this` signals that this sets an instance field, not a local. I think it helps readability when not in an IDE.

Reply:
Yes, I agree, but the next line contains pure {{footer}} and it initally mislead me to believe that that is a different {{footer}}.

Reply:
I think the rationale for this style is that when you see an access, you go looking for where the variable was assigned. That assignment should be clearly a local or instance field. Not using `this` for access is done because it is superfluous like you say.

Reply:
I see, thanks for the explanation. SGTM.